packages/nextjs/components/NetworksDropdown/AddCustomChainModal.tsx

```diff
@@ -0,0 +1,123 @@
+import { forwardRef } from "react";
+import { chainToOption, formDataToChain, getStoredChainsFromLocalStorage, storeChainInLocalStorage } from "./utils";
+import { Options } from "./utils";
+import { XMarkIcon } from "@heroicons/react/24/outline";
+import { useGlobalState } from "~~/services/store/store";
+import { notification } from "~~/utils/scaffold-eth";
+
+interface AddCustomChainModalProps {
+  groupedOptionsState: any; // @todo type anys
+  setGroupedOptionsState: React.Dispatch<React.SetStateAction<any>>;
+  setSelectedOption: React.Dispatch<React.SetStateAction<Options | null>>;
+  onChange: (option: Options | null) => void;
+}
```

Comment:
```ts
type AddCustomChainModalProps = {
  groupedOptionsState: GroupedOptions;
  setGroupedOptionsState: React.Dispatch<React.SetStateAction<GroupedOptions>>;
  setSelectedOption: React.Dispatch<React.SetStateAction<Options | null>>;
  onChange: (option: Options | null) => void;
};
```

Also in SE-2 codebase we always default to `type`  instead of `interface`, and use interface only when necessary